site/public/js/gradeable-notebook.js

```diff
@@ -130,46 +129,70 @@ function saveNotebookToLocal() {
-                                //check proper inputs
-                                for (let i = 0; i < answers.length; i++) {
-                                    if ($(this)[0].defaultValue === answers[i]) {
-                                        $(this).prop('checked', true);
-                                    }
-                                }
-                            });
-                        });
+            const {file_name, answers} = inputs.multiple_choice[id];
+            let found = false;
+            $(`fieldset.mc[data-filename="${file_name}"] input`).each(function(){
+                found = true;
+                //check off proper inputs
+                for (let i = 0; i < answers.length; i++) {
+                    if ($(this)[0].defaultValue === answers[i]) {
+                        $(this).prop('checked', true);
+                        $(this).trigger('change');
                     }
-                });
+                }
             });
+            if (!found) {
+                not_found.push(inputs.multiple_choice[id]);
+            }
         }
+
         //restore short answers
-        for (const id in inputs.short_answer) {
-            const filename = inputs.short_answer[id][0];
-            const answers = inputs.short_answer[id][1];
-            $('.short_answer').each(function(){
-                $(this).children('div').each(function(){
-                    //match file names
-                    if ($(this).attr('class') === filename) {
-                        //set input
-                        const editor = ($(this)[0]).querySelector('.CodeMirror').CodeMirror;
-                        editor.setValue(answers);
-                    }
-                });
-            });
+        for (const id in inputs.codebox) {
+            const {file_name, value} = inputs.codebox[id];
+            const question = $(`.short_answer > div[data-filename="${file_name}"]`);
+            //fill in proper values if question is found
+            if (question.length > 0) {
+                const editor = question[0].querySelector('.CodeMirror').CodeMirror;
+                editor.setValue(value);
+            }
+            else {
+                not_found.push(inputs.codebox[id][0]);
+            }
+        }
+
+        //if there are answers that could not be placed anywhere
+        if (not_found.length > 0) {
+            const old_answers_div = document.createElement('div');
+            old_answers_div.id = 'old-answers';
+            old_answers_div.classList.add('box');
+            old_answers_div.style.backgroundColor = 'var(--alert-background-danger-red)';
+
+            //create header text to warn user
+            const old_answers_header = document.createElement('h4');
+            old_answers_header.style.color = 'var(--focus-danger-red)';
+            old_answers_header.innerHTML = 'Answer(s) could not be restored you will have to copy and paste them in the proper place';
```

Comment:
I believe that the text here isn't grammatically correct.  Would adding a period after restored be better?
Suggestion: `Answer(s) could not be restored. You will have to copy and paste them in the proper place.`